lib/matplotlib/tests/test_image.py

```diff
@@ -1017,8 +1017,8 @@ def test_imshow_bool():
 def test_full_invalid():
     fig, ax = plt.subplots()
     ax.imshow(np.full((10, 10), np.nan))
-    with pytest.warns(UserWarning):
-        fig.canvas.draw()
+
```

Comment:
The previous warning was:
```bash
matplotlib/lib/matplotlib/image.py:442: UserWarning: Warning: converting a masked element to nan.
  vmid = np.float64(self.norm.vmin) + dv / 2
```
If I print out the previous `im.norm.vmin, im.norm.vmax`, they were both `np.ma.masked`, which stems from passing in a fully masked array (`np.max(np.ma.array([1, 2], mask=[True, True]))` returns `masked`), so here we are calling `np.float64(np.ma.masked)`. I don't think we want `masked` as vmin/vmax, so this perhaps even helps here... The vmin/vmax turn into `(0, 0)` now.

Note that the image doesn't change at all, just no warning about norm vmin/vmax anymore.

Reply:
This is fun second-order behavior change!

- In https://github.com/matplotlib/matplotlib/pull/6700 we stop passing vmin/vmax through `float` on init because it stripped precision of float128
- this broke yt that was expecting this to strip off units so we got https://github.com/matplotlib/matplotlib/pull/6700 to add `_sanitize_extrema` which _gently_ passes things through float
- this got updated in #12478 due to changes in numpy to deprecate np.asscalar 
- this PR now passes _all_ setting of vmin/vmax through `_sanitize_extrema`  (instead of previously just on init)
- `np.ma.masked.item() == 0` but `float(np.ma.masked) is np.nan`

Previously if you did 

```python
im = plt.imshow(np.full((10, 10), np.nan), vmin=np.ma.masked, vmax=np.ma.masked)
assert im.norm.vmin == 0
assert im.norm.vmax == 0
```

but if you did 

```python
im = plt.imshow(np.full((10, 10), np.nan)
assert im.norm.vmin is np.ma.masked
assert im.norm.vmax is np.ma.masked
```

(`==` is nan-like with `np.ma.masked`).

I would argue that this is a bug, in the sense that explicitly passing the `np.max(data)` and letting us auto-computer the max gave different results.  

We also say that if vmin == vmax the result of `__call__` will be 0.

This will lead to a subtle behavior change in the case where:
 - the user passed us all nan
 - let us infer the vmin/vmax
 - later updates the array with non-nan values
  
Previously the user would get all "bad" color, now they will get the "0" color.  Given the other difference in behavior of explicitly passing vmin/vmax, the agreement with "if vmin/vmax match -> go to 0", this feeling like a very cornery corner case, and the results already being useless from a communications point of view, I think we can leave this as-is and do not need a further API change note.